Kusama: remove migrations applied by `spec_version: 1_000_000`

acatangiu opened the pull request:
Runtimes 1.0.0 has been released: https://github.com/polkadot-fellows/runtimes/releases/tag/v1.0.0 and Kusama has now successfully updated on-chain to spec: `1_000_000`.

This PR removes migrations that have been applied by the successful update of the Kusama runtime to above version.

acatangiu commented:
The 1st commit is the mandatory one, it removes the `SessionKeys` migration which doesn't have protections in place to allow multiple runs of the migration.

The other ones' removal at this point is optional, but if we see them applied on-chain, let's also remove them.

bkchr commented on relay/kusama/src/lib.rs:
```diff
@@ -1657,108 +1620,10 @@ pub type Migrations = migrations::Unreleased;
 
 	/// Unreleased migrations. Add new ones here:
 	pub type Unreleased = (
 		init_state_migration::InitMigrate,
```
```suggestion
```

Can also be removed.

liamaharon reviewed (approved):
Why only remove from Kusama?

acatangiu commented:
> Why only remove from Kusama?

I'm thinking we do separate PRs that follow actual on-chain upgrades + migrations for each chain.

When kusama is upgraded and we can **verify** these migrations have been successful we'll merge PR to remove them.

My suggestions is same process, different PR+timeline for Polkadot.

acatangiu commented:
referenda for upgrading Kusama: https://kusama.subsquare.io/referenda/297

acatangiu commented:
Kusama now successfully updated to `spec: 1_000_000` - all these migrations have now run (and been finalized) on-chain so we can remove them.

Please note, we **have to remove** the session keys migration before the next release as it doesn't gracefully handle reapplying it.

acatangiu commented:
@andresilva please a rank-3 review here to get this in and avoid accidents like on Westend

acatangiu commented on relay/kusama/src/paras_scheduler_migration.rs:
```diff
@@ -104,6 +104,7 @@ pub mod v1 {
 		ValueQuery,
 	>;
 
+	#[allow(dead_code)]
 	#[allow(deprecated)]
 	pub type MigrateToV1<T> = VersionedMigration<
```
Instead of this, I can also remove this file completely, but thought it might be useful for future reference.. lemme know if you think otherwise.

ordian replied:
I'd just remove it. We have git history for future reference.

acatangiu commented:
hmm.. since this touches `relay/` files, it needs 4 rank 3 approvals

fellows, please review!

acatangiu commented:
/merge

acatangiu commented:
I just rebased and fixed conflicts. Someone with merge rights, pls take a look and merge before we keep adding new stuff and start confusing already applied migrations with incoming new ones.

andresilva commented:
/merge

acatangiu commented:
> 6 of 13 checks passed 

this got auto-merged after rebase without running all CI tests - any way we can see if these failed or simply didn't run or what? 😃

andresilva commented:
You can check the CI build on master after this got merged.
